block/internal/da/client.go

```diff
@@ -180,8 +182,35 @@ func (c *client) Submit(ctx context.Context, data [][]byte, _ float64, namespace
 	}
 }
 
+// getBlockTimestamp fetches the block timestamp from the DA layer header.
+// If the header fetch fails, it falls back to time.Now() and logs a warning.
+func (c *client) getBlockTimestamp(ctx context.Context, height uint64) time.Time {
+	if c.headerAPI == nil {
+		c.logger.Warn().Uint64("height", height).Msg("header API not available, using current time")
+		return time.Now()
+	}
+
+	headerCtx, cancel := context.WithTimeout(ctx, c.defaultTimeout)
+	defer cancel()
+
+	header, err := c.headerAPI.GetByHeight(headerCtx, height)
```

Comment:
can we return previous time (time.Time{}) instead? So we fail earlier.

Reply:
https://github.com/evstack/ev-node/pull/2939/changes/d8a31921985121875fe1744ac8f17ca9bb77889c